### PR TITLE
Replaced from and to argument with a month argument.

### LINKED ISSUE #14: 🐛 Bad request returned when end and start date are not the same
When I call the cli tool with values Start date: 2024-07-03 End date: 2024-07-31, the tool writes the error `Response status code does not indicate success: 400 (Bad Request)` to the .csv files.

If I run the tool with the --debug flag I can see the error as described [here](https://learn.microsoft.com/en-us/azure/carbon-optimization/troubleshooting#42-one-month-date-scope-supported-in-certain-api-report-types)
As we are using the type `ItemDetailsReport` the start and end date should be equal.

I propose we remove the from and to parameters as they are not useful right now.
Let's replace it with a `DateOnly` that captures the month (and the year) the user wants to receive information about.

### PR DESCRIPTION
**Pull request by JvandeKraats**

### CODE REVIEW

**Comment by JvandeKraats** [comment]
Removed the validation code because month can't be null and the from and to checks are not needed anymore.

File: src/Properties/launchSettings.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "profiles": {
     "azure-carbon-cli": {
       "commandName": "Project",
-      "commandLineArgs": "--subscription 2193e77b-d7ae-498b-9a28-14abbd97dfe2 --from 2024-05-01 --to 2024-05-01 --debug"
+      "commandLineArgs": "--subscription 2193e77b-d7ae-498b-9a28-14abbd97dfe2 --month 2024-07-01"
```
Comment (DannyvanderKraan) [inline src/Properties/launchSettings.json]: if we strip out the from and to parameters, maybe its better to make the command line parameters  --year and --month and if you leave --year out it'll just be the current year? This way you don't have to deal with the day part.

**Review by DannyvanderKraan — CHANGES_REQUESTED** [review]

File: src/Properties/launchSettings.json (hunk 1 above)
Comment (JvandeKraats) [reply src/Properties/launchSettings.json]: That's a good point. I will implement this

**Review by DannyvanderKraan — APPROVED** [review]
LGTM

**Review by DannyvanderKraan — COMMENTED** [review]
LGTM

**Pull request merged**